### PR TITLE
Sh Cas Improvements

### PR DESCRIPTION
**Pull request by stevenhammerton**
Realised I made an error in some conflict resolution.

Also added a configurable prefix to user ids.

### CODE REVIEW

**Comment by erikjohnston** [comment]
@matrixbot ok to test

File: synapse/config/cas.py [hunk 1]
```diff
@@ -27,13 +27,16 @@ def read_config(self, config):
         if cas_config:
             self.cas_enabled = True
             self.cas_server_url = cas_config["server_url"]
+            self.cas_user_id_prefix = cas_config["user_id_prefix"]
```
Comment (erikjohnston) [inline synapse/config/cas.py]: Might it be worth making this optional? And defaulting to the empty string?

Comment (stevenhammerton) [inline synapse/config/cas.py]: I'd argue it's safer to be required with empty string supported but requires explicit enablement. That way, the likelihood of enabling CAS and that allowing someone else to log in as an already created user is reduced.

**Comment by erikjohnston** [comment]
@stevenhammerton Yeah, I fixed the bad merge yesterday :)  It might be worth getting [sytest](https://github.com/matrix-org/sytest) working locally. Currently we only automatically run sytest against develop.

Otherwise, LGTM

**Comment by erikjohnston** [comment]
Although, it may be worth making the CAS user prefix immutable; otherwise I can foresee all sorts of confusion arising from people changing it thinking it would automagically do the right thing.

**Comment by stevenhammerton** [comment]
Yes, good point. Maybe this requires more thinking. I'll close the PR for now as the conflict error has already been fixed.

**Comment by erikjohnston** [comment]
Cool. I think we make the `server_name` config option immutable by making a one row table and storing it in the DB (or maybe it just looks at the users table). If you go down that route it might be useful to make a generic "immutable config options" table that maps from `key -> value`, or something.

File: synapse/config/cas.py (hunk 1 above)
Comment (erikjohnston) [inline synapse/config/cas.py]: Mmm, true.

**Pull request closed without merging**